### PR TITLE
151629729 enable social logins

### PR DESCRIPTION
**Pull request by emugaya**
#### What does this PR do?
This PR Enable users to Login / Signin `wger` using `Facebook`, `Google`, and `Twitter`.

#### Description of Task to be completed?
Users are now able to access `wger`  using social media. Users no longer have to create accounts to use wger making it easy for them to signup and use the application.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/151629729

####  Screenshot
image

### CODE REVIEW

File: wger/core/static/css/bootstrap-custom.css [hunk 1]
```diff
@@ -20,7 +20,7 @@
  * Adapt some of bootstrap's CSS classes and styles
  *
  **************************************************************/
-
+@import url("../bootstrap-social-gh-pages/css/bootstrap.css");
```
Comment (aenkya) [inline wger/core/static/css/bootstrap-custom.css]: What's the use of the bootstrap import?

**Review by aenkya — COMMENTED** [review]
LGTM

File: wger/core/templates/user/login.html [hunk 2]
```diff
@@ -7,30 +7,63 @@
 {% block header %}
 {% endblock %}
 
+
```
Comment (margierain) [inline wger/core/templates/user/login.html]: remove space .. not useful

File: wger/core/urls.py [hunk 3]
```diff
@@ -39,7 +39,7 @@
 
 # sub patterns for user
 patterns_user = [
-    url(r'^login$', user.login, name='login'),
+    url(r'^login', user.login, name='login'),
```
Comment (margierain) [inline wger/core/urls.py:42]: why is the $ sign removed

File: wger/settings_global.py [hunk 4]
```diff
@@ -128,11 +132,18 @@
...
+    'social_django.middleware.SocialAuthExceptionMiddleware',
 )
 
 AUTHENTICATION_BACKENDS = (
+    'wger.utils.helpers.EmailAuthBackend',
+    'social_core.backends.open_id.OpenIdAuth',
+    'social_core.backends.google.GoogleOAuth2',
+    'social_core.backends.twitter.TwitterOAuth',
+    'social_core.backends.facebook.FacebookOAuth2',
     'django.contrib.auth.backends.ModelBackend',
```
Comment (margierain) [inline wger/settings_global.py]: 'django.contrib.auth.backends.ModelBackend' should be the first thing, the order is important, the auth model should run first

File: wger/settings_global.py [hunk 5]
```diff
@@ -396,3 +410,35 @@
...
+SOCIAL_AUTH_FACEBOOK_API_VERSION = '2.10'
+SOCIAL_AUTH_FACEBOOK_SCOPE = ['email']
+
+SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = os.environ.get("SOCIAL_AUTH_GOOGLE_OAUTH2_KEY")
+SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = os.environ.get("SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET")
+SOCIAL_AUTH_GOOGLE_OAUTH2_SCOPE = ['email']
+
+SOCIAL_AUTH_LOGIN_ERROR_URL = '/en/user/login/'
+SOCIAL_AUTH_RAISE_EXCEPTIONS = False
+DEBUG = False
```
Comment (margierain) [inline wger/settings_global.py]: repetition.. check line 29... attention to detail

**Review by margierain — CHANGES_REQUESTED** [review]
left comments kindly review

File: wger/settings_global.py (hunk 5 above)
Comment (emugaya) [reply wger/settings_global.py]: Removed

File: wger/settings_global.py (hunk 4 above)
Comment (emugaya) [reply wger/settings_global.py]: Order adjusted

File: wger/core/urls.py (hunk 3 above)
Comment (emugaya) [reply wger/core/urls.py:42]: $ Added. This was removed to manage some errors during redirection.

File: wger/core/templates/user/login.html (hunk 2 above)
Comment (emugaya) [reply wger/core/templates/user/login.html]: Space Removed

File: wger/core/static/css/bootstrap-custom.css (hunk 1 above)
Comment (emugaya) [reply wger/core/static/css/bootstrap-custom.css]: it was intended to import bootstrap social login icons. I instead copied the specific CSS. It has been removed

**Pull request merged**